quickwit/quickwit-config/src/index_config/serialize.rs

```diff
@@ -0,0 +1,258 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::str::FromStr;
+
+use anyhow::{bail, Context};
+use quickwit_common::uri::Uri;
+use serde::de::Error;
+use serde::{Deserialize, Deserializer, Serialize};
+use tracing::info;
+
+use crate::{
+    build_doc_mapper, validate_identifier, ConfigFormat, DocMapping, IndexConfig, IndexingSettings,
+    RetentionPolicy, SearchSettings,
+};
+
+/// Deserializes and validates a [`Uri`].
+fn deser_and_validate_uri<'de, D>(deserializer: D) -> Result<Option<Uri>, D::Error>
```

Comment:
Now `Uri` implements `Deserialize`. We can deserialize directly into `Option<Uri>` and remove this function.

Reply:
nice